src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +102,1361 @@ static DamageTable D_80BBF858 = {
+}
+
+void EnSGoro_ActionAsleepMessageWait(EnSGoro* this, PlayState* play) {
+    if (Message_GetState(&play->msgCtx) == 6) {
+        if (Message_ShouldAdvance(play)) {
+            play->msgCtx.msgMode = 0x43;
+            play->msgCtx.unk12023 = 4;
+            this->actionFunc = EnSGoro_ActionAsleep;
+        }
+    }
+    EnSGoro_UpdateSleeping(this, play);
+}
 
-extern ColliderCylinderInit D_80BBF820;
-extern CollisionCheckInfoInit2 D_80BBF84C;
-extern DamageTable D_80BBF858;
+void EnSGoro_Init(Actor* thisx, PlayState* play) {
 
-extern UNK_TYPE D_060091A8;
-extern UNK_TYPE D_06012DE0;
+    EnSGoro* this = THIS;
+    SkelAnime* skelAnime = &this->skelAnime;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBCA80.s")
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 20.0f);
+    SkelAnime_InitFlex(play, skelAnime, &gGoronSkel, &gGoronUnrollAnim, this->jointTable, this->morphTable,
+                       GORON_LIMB_MAX);
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    if (this->actor.update != NULL) {
+        s32 objIndex = Object_GetIndex(&play->objectCtx, OBJECT_OF1D_MAP);
+        this->loadedObjIndex = objIndex;
+        if (objIndex < 0) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    }
+    this->actor.draw = EnSGoro_Draw;
+    this->actionFunc = EnSGoro_ActionDefault;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBD348.s")
+void EnSGoro_Destroy(Actor* thisx, PlayState* play) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBD8F0.s")
+    EnSGoro* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBD93C.s")
+    Collider_DestroyCylinder(play, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBD98C.s")
+void EnSGoro_Update(Actor* thisx, PlayState* play) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBDACC.s")
+    EnSGoro* this = (EnSGoro*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBDC34.s")
+    this->actionFunc(this, play);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 30.0f, 12.0f, 0.0f, 5U);
+    gSegments[6] = PHYSICAL_TO_VIRTUAL(play->objectCtx.status[this->loadedObjIndex].segment);
+    SkelAnime_Update(&this->skelAnime);
+    if (this->animInfoIndex != ENSGORO_ANIM_SLEEPY) {
+        EnSGoro_UpdateAttentionTarget(this, play);
+    }
+    EnSGoro_UpdateEyes(this);
+    EnSGoro_UpdateActorFocus(this);
+    EnSGoro_UpdateCollider(this, play);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBDCFC.s")
+s32 EnSGoro_CalculateTranslateRotate(s16 amt_z, s16 amt_y, Vec3f* translate, Vec3s* rotate, s32 delay_mode,
+                                     s32 offset_rotate) {
```

Comment:
This looks like `SubS_UpdateLimb`. Personally I would prefer to have the same variable names and a similar function name.